### ISSUE TITLE
[FR]Expand and collapse child node connections in mind map style

### ISSUE THREAD

**Issue by PennHui666** [Developer-Mike/obsidian-advanced-canvas#21]
## Issue Summary

Regarding the application of the "mind map" mode in canvas, there are already projects that have made relevant features.
[Obsidian-Canvas-MindMap](https://github.com/Quorafind/Obsidian-Canvas-MindMap)
* Press Enter to create brother card;
* Press Tab to create child card;
* Delete card to rearrange layout automatically;
* Press Alt + ←/→/↓/↑ to navigate between cards;
* Press Cmd / Ctrl + ←/→/↓/↑ to create floating cards;

But it’s not enough. I think this project can further enhance the idea of displaying the canvas mind map mode by emulating the collapsing/expanding features implemented on v2.6.0-beta.1.

## Feature Description

- Collapse of child nodes by clicking the expand (⊕) or collapse (⊖) symbols next to the father node.
- Display the number of collapsed child nodes after collapse
- 

## Example
![exam](https://github.com/Developer-Mike/obsidian-advanced-canvas/assets/72249772/83d9a6df-6bc7-4c8c-89bf-509b4abd6f66)

**Comment by PennHui666** [comment]
btw，maybe it would be simpler to do this by double-clicking the connection point on the card, rather than adding another icon,if this is possible.

**Comment by net-frog** [comment]
I am using groups to simulate this, but there is one small problem that if fold and move the folded group close to other groups and when you unfold they overlap with other group. Maybe, when unfolding if there are other cards or group it should be moved automatically.

**Comment by Francespo** [comment]
Why isn't this planned anymore?

**Comment by Developer-Mike** [comment]
It's still planned. I've restructured the labels of this GitHub repositiory and there isn't a "planned" label anymore (Because nearly every open FR is planned).

**Comment by AndreiPashkin** [comment]
What do you guys think about integrating existing layout engines (like [ELK](https://github.com/kieler/elkjs) or [dagre](https://github.com/dagrejs/dagre)) into the app?

**Comment by Developer-Mike** [comment]
Yes, both layout engines are being looked at. But elkjs is currently my favorite of them. Is there a specific reason for you to ask?

**Comment by Mpic** [comment]
I would also add that it would be ideal if the wonderful Floating edges (auto edge side) feature could be automatic for the nodes created with the "mindmap mode".

**Comment by Developer-Mike** [comment]
@HKohlhoff Thanks for developing the Obsidian plugin. This is an issue/FR about Advanced Canvas though. Next time you can ask me if it's okay first.

If someone is interested in collapsing outgoing edges, check out the [Canvas Folding](https://github.com/HKohlhoff/canvas-folding) plugin by HKohlhoff.

**Comment by HKohlhoff** [comment]
@Developer-Mike:
Sorry I did not mean to make advertisements in your repo. Deleted all of it.
